src/compiler/checker.ts

```diff
@@ -4632,6 +4659,43 @@ module ts {
                     }
                     result &= related;
                 }
+
+                if (source.typePredicate && target.typePredicate) {
+                    let hasDifferentParamaterIndex = source.typePredicate.parameterIndex !== target.typePredicate.parameterIndex;
+                    let hasDifferentTypes = !isTypeIdenticalTo(source.typePredicate.type, target.typePredicate.type);

```

Comment:
If you don't report both errors (which I've already explained), then you can defer this until you know the indices are equal.